common/crypto/src/asymmetric/identity/mod.rs

```diff
@@ -87,15 +102,31 @@ impl KeyPair {
         Ok(KeyPair {
             private_key: PrivateKey::from_bytes(priv_bytes)?,
             public_key: PublicKey::from_bytes(pub_bytes)?,
+            index: fake_index(pub_bytes),
```

Comment:
but the resultant `index` here will be different when you call `KeyPair::from_bytes` and `KeyPair::from_secret` even if you use the same underlying keypair - is that intended?

Reply:
Yeah, this is not a derived key, we just need something that is not None or a constant, in case something somewhere inspects the keys and behaves differently if it can figure out they're derived